### PR TITLE
Try to regenerate the dependency tree that lives in the solution

### PR DESCRIPTION
**Pull request by DHowett-MSFT**
- An attempt to fix parallel builds emerges.

### CODE REVIEW

**Comment by DHowett-MSFT** [comment]
This will fail, but hopefully the failures will point at more missing dependencies instead of leading us in circles. I have a good feeling about the upcoming failures' usefulness.

**Comment by DHowett-MSFT** [comment]
I've reverted Michael's build fix. The bulk of the change here is putting a dependency graph in the solution _and_ putting project GUIDs in every project reference. :smile:

File: src/cascadia/TerminalApp/TerminalApp.vcxproj [hunk 1]
```diff
@@ -60,31 +60,32 @@
     <ProjectReference Include="$(OpenConsoleDir)src\types\lib\types.vcxproj">
       <Project>{18D09A24-8240-42D6-8CB6-236EEE820263}</Project>
     </ProjectReference>
-
     <!-- The midl compiler however, _will_ aggregate our winmd dependencies
     somehow. So make sure to only include top-level dependencies here (don't
     include Settings and Connection, since Control will include them for us) -->
-    <ProjectReference Include="$(OpenConsoleDir)src\cascadia\TerminalSettings\TerminalSettings.vcxproj" />
```
Comment (zadjii-msft) [inline src/cascadia/TerminalApp/TerminalApp.vcxproj]: this makes me sad it's no longer supported

**Pull request closed without merging**